lib/compat/wordpress-6.3/class-gutenberg-rest-global-styles-controller-6-3.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Global Styles REST API Controller.
+ */
+class Gutenberg_REST_Global_Styles_Controller_6_3 extends Gutenberg_REST_Global_Styles_Controller_6_2 {
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @since 6.3 Registers `/revisions` endpoint.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\/\w-]+)/revisions',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item_revisions' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'id' => array(
+							'description'       => __( 'The id of the global styles post.', 'gutenberg' ),
+							'type'              => 'string',
+							'sanitize_callback' => array( $this, '_sanitize_global_styles_callback' ),
+						),
+					),
+				),
+			)
+		);
+		parent::register_routes();
+	}
+
+	/**
+	 * Returns revisions of the given global styles config custom post type.
+	 *
+	 * @since 6.3
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item_revisions( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+		$revisions                        = array();
+		$raw_config                       = json_decode( $post->post_content, true );
+		$is_global_styles_user_theme_json = isset( $raw_config['isGlobalStylesUserThemeJSON'] ) && true === $raw_config['isGlobalStylesUserThemeJSON'];
+
+		if ( $is_global_styles_user_theme_json ) {
+			$user_theme_revisions = wp_get_post_revisions(
+				$post->ID,
+				array(
+					'posts_per_page' => 100,
+				)
+			);
+
```

Comment:
If empty, return a 404 error?

Reply:
It did cross my mind, but I can't find a precedent for this in WordPress. 

[WP_REST_Revisions_Controller](https://github.com/ramonjd/wordpress-develop/blob/b24f990f4f7789cee65628ef090a9e68a2798b5b/src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L222-L222) returns an empty array where no revisions are found I believe. What do you reckon?